src/Database/DatabaseConfig.php

```diff
@@ -114,6 +121,37 @@ public function connection(): array
         );
     }
 
+    /**
+     * Tenant's host database connection config.
+     */
+    public function hostConnection(): array
+    {
+        $config = $this->tenantConfig();
+        $template = $this->getTemplateConnectionName();
+        $templateConnection = config("database.connections.{$template}");
+
+        if ($this->manager() instanceof Contracts\ManagesDatabaseUsers) {
+            // We're removing the username and password because user with these credentials is not created yet
+            unset($config['username']);
+            unset($config['password']);
+        }
+
+        if (! $config) {
+            return $templateConnection;
+        }
+
+        return array_replace($templateConnection, $config);
+    }
+
+    /**
+     * Purge host database connection.
+     */
+    public function purgeHostConnection(): void
+    {
+        $tenantHostConnectionName = $this->getTenantHostConnectionName();
+        DB::purge($tenantHostConnectionName);
```

Comment:
```suggestion
        DB::purge($this->getTenantHostConnectionName());
```

Also I'd add a short comment explaining what this is for (similar to what we discussed in the reviews). Should say something along the lines of: this is used for cleaning up the connection before creating it for the current tenant.

Reply:
Added the comment.